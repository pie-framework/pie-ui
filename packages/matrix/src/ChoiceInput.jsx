import React from 'react';
import PropTypes from 'prop-types';
import {withStyles} from '@material-ui/core/styles';

import {color} from '@pie-lib/render-ui';
import Radio from '@material-ui/core/Radio';

const radioStyles = {
  root: {
    color: `var(--choice-input-color, ${color.text()})`
  },
  checked: {
    color: `var(--choice-input-selected-color, ${color.text()})`
  }
};

const ChoiceInput = withStyles(radioStyles)((props) => {
  const {disabled, checked, matrixKey, matrixValue, classes, onChange} = props;

  const onChangeWrapper = () => {
    onChange({
      matrixValue,
      matrixKey
    });
  };

  return (
    <Radio
      checked={checked}
      onChange={onChangeWrapper}
      disabled={disabled}
      classes={{
        root: classes.root,
        checked: classes.checked,
      }}
    />
  );
});

ChoiceInput.propTypes = {
  checked: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
  matrixValue: PropTypes.number.isRequired,
  matrixKey: PropTypes.string.isRequired,
  classes: PropTypes.object,
  onChange: PropTypes.func.isRequired
};

ChoiceInput.defaultProps = {
  checked: false,
  disabled: false
};

export default ChoiceInput;
