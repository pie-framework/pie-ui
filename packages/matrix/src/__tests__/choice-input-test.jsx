import React from 'react';
import {shallow, mount} from 'enzyme';
import ChoiceInput from '../ChoiceInput';
import toJson from 'enzyme-to-json';

describe('ChoiceInput', () => {

  const getPropsDefault = (onChange, propsToOverride) => {
    return {
      checked: false,
      disabled: false,
      matrixValue: 2,
      matrixKey: 'matrixKey',
      classes: {
        root: 'root',
        checked: 'checked',
      },
      onChange,
      ...propsToOverride
    }
  };

  describe('snapshots', () => {
    it('renders', () => {
      const onChange = jest.fn();
      const props = getPropsDefault(onChange);
      const shallowInstance = shallow(<ChoiceInput {...props} />);
      expect(toJson(shallowInstance)).toMatchSnapshot();
    });

    it('calls onChange', () => {
      const onChange = jest.fn();
      const props = getPropsDefault(onChange);

      const mountInstance = mount(<ChoiceInput {...props} />);

      mountInstance.find('input').at(0).simulate('change', {
        target: {
          id: 'name',
          value: '',
        },
      });

      expect(onChange).toHaveBeenCalledTimes(1);
    });

    it('does not calls onChange if disabled', () => {
      const onChange = jest.fn();
      const props = getPropsDefault(onChange, {disabled: true});

      const mountedComponent = mount(<ChoiceInput {...props} />);

      mountedComponent.find('input').at(0).simulate('change', {
        target: {
          id: 'name',
          value: '',
        },
      });

      expect(onChange).toHaveBeenCalledTimes(0);
    });
  });
});
