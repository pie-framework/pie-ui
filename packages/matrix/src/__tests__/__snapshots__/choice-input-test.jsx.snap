// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoiceInput snapshots renders 1`] = `
<Component
  checked={false}
  classes={
    Object {
      "checked": "Component-checked-2 checked",
      "root": "Component-root-1 root",
    }
  }
  disabled={false}
  matrixKey="matrixKey"
  matrixValue={2}
  onChange={[MockFunction]}
/>
`;
